src/main/java/use_cases/AddUseCase.java

```diff
@@ -16,19 +16,15 @@ public AddUseCase(AddOutputBoundary presenter){
      * @param data a Data Structure containing the serial number and quantity of the item to add
      * @return a string for the view to show the user (created by the presenter)
      */
-    public String addItem(AddDS data){
-
-        if (data.getQuantity() < 0){
-            return presenter.prepareFailure(0);
-        } // if the employee enters a negative quantity (unlikely, but you don't want this to happen at all)
-        else if (! TempDataStorage.hasItem(data.getSerialNum())){
-            return presenter.prepareFailure(1);
-        } // can remove this later, assuming that serial number entered is never incorrect - just kept in for now for testing purposes
+    public void addItem(AddDS data){
 
```

Comment:
I think we discussed later that it would be better if this functionality is kept within the addUseCase in order to implement MVC model. Also, if the code is moved back to addUseCase, the call to the presenter must be made through the interface that the addPresenter implements as otherwise it violates clean architecture.

Reply:
You're right. I've made the necessary changes and added them in a new commit.